apps/scorecard/server/dataFunctions/processScorecard.js

```diff
@@ -173,95 +177,125 @@ processScorecard = function (plotParams, plotFunction) {
-      // or they are handled individually, so we remove the ones that are handled individually from the single select list.
-      const notIncludedParams = ["label", "data-source", "validation-data-source"];
-      const rowParameters = singleCurveParamNames.filter(function (paramName) {
-        return !notIncludedParams.includes(paramName);
-      });
+  scorecardDocument.results.title = title;
+  scorecardDocument.results.rows = {};
+  // fill in the rows - these are all initially default values
+  plotParams.curves.forEach(function (curve) {
+    // create the empty object for this row
+    const { label } = curve;
+    scorecardDocument.results.rows[label] = {};
+    // add the top level elements.
+    // make rowTitle and rowParameters maps, the display page can sort out stringifying them.
+    // map the necessary row parameters
+    // remove these params from the singleCurveParamNames list, all of the row parameters are either single select
+    // or they are handled individually, so we remove the ones that are handled individually from the single select list.
+    const notIncludedParams = ["label", "data-source", "control-data-source"];
+    const rowParameters = singleCurveParamNames.filter(function (paramName) {
+      return !notIncludedParams.includes(paramName);
+    });
 
-      scorecardDocument.results.rows[curve.label].rowTitle = {
-        label,
-        datasource: curve["data-source"],
-        validationDatasource: curve["validation-data-source"],
-      };
-      scorecardDocument.results.rows[curve.label].rowParameters = rowParameters;
-      scorecardDocument.results.rows[curve.label].regions = regions;
-      scorecardDocument.results.rows[curve.label].fcstlens = fcstLengths;
-      scorecardDocument.results.rows[curve.label].data = {};
-      regions.forEach(function (region) {
-        if (scorecardDocument.results.rows[curve.label].data[region] === undefined) {
-          scorecardDocument.results.rows[curve.label].data[region] = {};
+    scorecardDocument.results.rows[curve.label].rowTitle = {
+      label,
+      dataSource: curve["data-source"],
+      controlDataSource: curve["control-data-source"],
+    };
+    scorecardDocument.results.rows[curve.label].rowParameters = rowParameters;
+    scorecardDocument.results.rows[curve.label].regions = regions;
+    scorecardDocument.results.rows[curve.label].fcstlens = fcstLengths;
+    scorecardDocument.results.rows[curve.label].data = {};
+    curve.threshold =
+      curve.threshold === undefined ? ["threshold_NA"] : curve.threshold;
+    curve.level = curve.level === undefined ? ["level_NA"] : curve.level;
+    regions.forEach(function (region) {
+      region = region.replace(/\./g, "__DOT__");
+      if (scorecardDocument.results.rows[curve.label].data[region] === undefined) {
+        scorecardDocument.results.rows[curve.label].data[region] = {};
+      }
+      curve.statistic.forEach(function (stat) {
+        stat = stat.replace(/\./g, "__DOT__");
```

Comment:
Since there are a number of these in different places in this file - I think these `.replace(/\./g, "__DOT__")` calls should be factored out into a function so that we have one place to handle & update this in the code. Otherwise, we're going to have to be careful to not miss any instances or make any typos if we make changes in the future. 

This also becomes an easier piece of code to test if we get into unit testing.

e.g.

```js
/** A function to sanitize JSON keys by replacing the "." character with "__DOT__" */
function sanitizeKeys(str) {
   const regex = /\./g;
   return str.replace(regex, "__DOT__");
};

// usage
stat = sanitizeKeys(stat);
```